### PR TITLE
Move from env variables to AWS SecretsManager

### PR DESCRIPTION
**Pull request by Pinioo**

### CODE REVIEW

File: coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
+        print('Error: CoioteDM credentials are not set up correctly')
+        raise EnvironmentError()
+
+    secretsMap = json.JSONDecoder().decode(getSecretValueResponse['SecretString'])
+
+    credentials = CoioteCredentials(
+        url=secretsMap['url'],
+        username=secretsMap['username'],
+        password=secretsMap['password']
+    )
```
Comment (Tequilac) [inline coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py:34]: Handle errors by returning them with status code.

File: coiote-aws-iot-terraform/main.tf [hunk 2]
```diff
@@ -15,6 +15,22 @@ resource "aws_iam_role" "iam_for_lambda" {
   ]
 }
 EOF
+
+  inline_policy {
+    name = "SecretsManagerForLambdaPolicy"
+
+    policy = jsonencode({
```
Comment (Tequilac) [inline coiote-aws-iot-terraform/main.tf:22]: Do the same with the policy above if possible

File: coiote-aws-iot-cloud-formation/cloudFormation.json [hunk 3]
```diff
@@ -16,6 +16,22 @@
             }
           ]
         },
+        "Policies": [
+          {
+            "PolicyName": "SecretsManagerForLambdaPolicy",
```
Comment (Tequilac) [inline coiote-aws-iot-cloud-formation/cloudFormation.json:21]: Won't the names collide when we create the role the second time?

Comment (Tequilac) [reply coiote-aws-iot-cloud-formation/cloudFormation.json:21]: https://stackoverflow.com/questions/57660653/how-to-create-some-random-or-unique-value-in-a-cloudformation-template

File: coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py [hunk 4]
```diff
@@ -1,10 +1,53 @@
...
+    urlKey = 'url'
+
+    expectedKeysSet = {usernameKey, passwordKey, urlKey}
+    fetchedKeysSet = set(secretsMap.keys())
+
+    if expectedKeysSet.issubset(fetchedKeysSet):
+        return CoioteCredentials(
+            url=secretsMap['url'],
+            username=secretsMap['username'],
+            password=secretsMap['password']
```
Comment (Tequilac) [inline coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py:43]: You can use the key strings created earlier

**Review by Tequilac — APPROVED** [review]

File: coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py (hunk 4 above)
Comment (Pinioo) [reply coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py:43]: That was a point of creating them, I just forgot to change it. Thanks.

**Pull request merged**